### PR TITLE
Updating 'openssh' to 8.5p1 to fix CVE-2021-28041.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Our current version is too far behind to backport the fix, so I'm updating `openssh` to the version, which is already free of the vulnerability.

###### Change Log  

- Updated `openssh` to version 8.5p1.
- Removed CVE (no)patches, which are already covered by the newer version.
- Removed regression test patch, since the newer version also already contains the right fix.

###### Does this affect the toolchain?  

No.

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2021-28041

###### Test Methodology

- Local package build.

### CODE REVIEW

File: cgmanifest.json [hunk 1]
```diff
@@ -3855,8 +3855,8 @@
         "type": "other",
         "other": {
           "name": "openssh",
-          "version": "8.0p1",
-          "downloadUrl": "https://ftp.usa.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-8.0p1.tar.gz"
+          "version": "8.5p1",
+          "downloadUrl": "https://ftp.usa.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-8.5p1.tar.gz"
```
Comment (jslobodzian) [inline cgmanifest.json:3859]: is it still p1?

File: SPECS/openssh/openssh.spec [hunk 2]
```diff
@@ -1,8 +1,8 @@
 %define systemd_units_rel 20191026
 Summary:        Free version of the SSH connectivity tools
 Name:           openssh
-Version:        8.0p1
-Release:        13%{?dist}
+Version:        8.5p1
```
Comment (jslobodzian) [inline SPECS/openssh/openssh.spec:4]: I didn't think it was still p1.

**Review by jslobodzian — APPROVED** [review]
:shipit:

File: cgmanifest.json (hunk 1 above)
Comment (PawelWMS) [reply cgmanifest.json:3859]: Discussed offline: yup, that's the way they still do it.

**Review by thcrain-msft — APPROVED** [review]

**Pull request merged**